### PR TITLE
TR-36: Remove duplicated typing in jira-oidc-helpers

### PR DESCRIPTION
**Pull request by binaryberserker**
remove unnecessary typings and cleanup in jira-oidc-helpers

### CODE REVIEW

**Comment by binaryberserker** [comment]
FYI indentation was kind of all over the place in jira-oidc-helpers, standardized it on tabs but could be worth a discussion at some point

**Comment by shoumaw** [comment]
> FYI indentation was kind of all over the place in jira-oidc-helpers, standardized it on tabs but could be worth a discussion at some point

Yeah we probably should create a ticket to integrate prettier with our current eslint config to ensure code formatting consistency across the code @justinbmeyer

**Comment by shoumaw** [comment]
The ticket's AC mentions something about

> Try to figure out a solution for the “meta” functions issue

I do not have much context about it, has it been covered in this PR?

**Comment by binaryberserker** [comment]
> The ticket's AC mentions something about
> 
> > Try to figure out a solution for the “meta” functions issue
> 
> I do not have much context about it, has it been covered in this PR?

hmm, ya i don't remember either, @justinbmeyer does this look something like what you were expecting? Personally i'd like to see these broken up some more rather than one giant collection of functions being returned from another function, but idunno how much of that we wana try to do in this ticket vs another and keep it incremental

File: public/shared/types.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
-type JsonResponse = {
+import { JiraIssue } from "../jira/shared/types"
+
+type JsonResponse <T>= {
```
Comment (binaryberserker) [inline public/shared/types.ts:3]: made JsonResponse generic

File: public/jira-oidc-helpers/index.ts
Comment (binaryberserker) [inline public/jira-oidc-helpers/index.ts:1]: broke up the `jira-oidc-helpers.ts` file into multiple files to hopefully make it easier to navigate. Put them into a `jira-oidc-helpers` folder with the `createJiraHelpers` function in the index to keep things fairly backwards compatible

File: public/jira-oidc-helpers/jira.ts
Comment (binaryberserker) [inline public/jira-oidc-helpers/jira.ts:1]: this contains most of the functions for working with jira, except for a couple of the larger functions that i moved into their own files

**Comment by binaryberserker** [comment]
> The ticket's AC mentions something about
> 
> > Try to figure out a solution for the “meta” functions issue
> 
> I do not have much context about it, has it been covered in this PR?

hmm, honestly I'm not sure what that's referring to either, I'll have to check on that

**Comment by binaryberserker** [comment]
also the changes from main should all be included as well, tho it wasn't a 1:1 relationship to keep all of them since the original `jira-oidc-helpers.ts` is deleted and the way the new one works is slightly different, so it may not be immediately obvious where those changes from main ended up, but they should all be there. Leme know if there's any you wana check for and i can show you where it ended up.

File: public/jira-oidc-helpers/storage.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+export function clearAuthFromLocalStorage() {
+    window.localStorage.removeItem("accessToken");
+    window.localStorage.removeItem("refreshToken");
+    window.localStorage.removeItem("expiryTimestamp");
+}
+
+export function fetchFromLocalStorage(key: string) {
+    return window.localStorage.getItem(key);
```
[the remark this answers is not in the document]
Comment (DavidNic11) [reply public/jira-oidc-helpers/storage.ts:33]: nah, that's separate from local storage

File: public/jira-oidc-helpers/jira.ts [hunk 3]
```diff
@@ -0,0 +1,345 @@
...
+    if (params.jql) formattedParams.jql = params.jql;
+    if (params.startAt) formattedParams.startAt = params.startAt.toString();
+    if (params.maxResults)
+        formattedParams.maxResults = params.maxResults.toString();
+    if (params.fields) formattedParams.fields = params.fields.join(",");
+    return formattedParams;
+}
+export function fetchIssueTypes(config: Config) {
+    return () => {
+        const response = config.requestHelper(`/api/3/issuetype`) as unknown;
```
[the remark this answers is not in the document]
Comment (DavidNic11) [reply public/jira-oidc-helpers/jira.ts:101]: https://github.com/bitovi/jira-timeline-report/pull/155#discussion_r1793861052

**Review by DavidNic11 — APPROVED** [review]

**Pull request merged**